### DISCUSSION TITLE
C++/CLI Support

### DISCUSSION THREAD

**Discussion by AliShahesmaeili** [dotnet/maui#498] [Q&A]
.NET Core 3.1 supports C++/CLI only on Windows
In MAUI with .NET 6  can we use the .NET interface through C++/CLI and C++/CX on mobile?

**Comment by Happypig375** [comment]
C++/CLI is Windows-only. Why would you expect MAUI to change that?

**Reply by pcarret** [reply]
I suppose it can be a legitimate question for those people having a huge c++ and crossplatform codebase to wonder about a solution to use their c++ codebase in conjunction with MAUI ?
It can be a CAD kernel for example like OpenCascade

**Comment by mwwm1** [comment]
[](https://csr.neom.com/dashboard/submissions)

**Comment by GUELIANEBelkacem** [comment]
Honestly this would be a complete game changer for us, having access to our C++ libraries without all the P Invoke gunk would probably be a huge selling point for MAUI, although the MAUI team is not the right people to be soliciting i suppose.
